### PR TITLE
doc/architecture: edit "Peering and Sets"

### PR DESCRIPTION
**Pull request by zdover23**
Edit the English in the section "Peering and Sets" in the file doc/architecture.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
inb4: I know that this section is likely not as clear to the uninitiated as it perhaps should be.

File: doc/architecture.rst [hunk 1]
```diff
@@ -615,46 +615,46 @@ Peering and Sets
...
 In previous sections, we noted that Ceph OSD Daemons check each other's
-heartbeats and report back to the Ceph Monitor. Another thing Ceph OSD daemons
-do is called 'peering', which is the process of bringing all of the OSDs that
-store a Placement Group (PG) into agreement about the state of all of the
-objects (and their metadata) in that PG. In fact, Ceph OSD Daemons `Report
-Peering Failure`_ to the Ceph Monitors. Peering issues  usually resolve
-themselves; however, if the problem persists, you may need to refer to the
-`Troubleshooting Peering Failure`_ section.
+heartbeats and report back to Ceph Monitors. Ceph OSD daemons also do something
+called 'peering', which is the process of bringing all of the OSDs that store a
```
Comment (anthonyeleven) [inline doc/architecture.rst:619]: OSD daemons also "peer", which is the

File: doc/architecture.rst [hunk 2]
```diff
@@ -615,46 +615,46 @@ Peering and Sets
...
+Placement Group (PG) into agreement about the state of all of the objects (and
```
Comment (anthonyeleven) [inline doc/architecture.rst:620]: RADOS objects

File: doc/architecture.rst [hunk 3]
```diff
@@ -615,46 +615,46 @@ Peering and Sets
...
+their metadata) in that PG. Ceph OSD Daemons `Report Peering Failure`_ to the
+Ceph Monitors. Peering issues usually resolve themselves; however, if the
+problem persists, you may need to refer to the `Troubleshooting Peering
+Failure`_ section.
 
-.. Note:: Agreeing on the state does not mean that the PGs have the latest contents.
+.. Note:: PGs that agree on the state of the cluster do not necessarily have
+   the current cluster contents.
```
Comment (anthonyeleven) [inline doc/architecture.rst:627]: s/cluster contents/data yet/ ?

File: doc/architecture.rst [hunk 4]
```diff
@@ -615,46 +615,46 @@ Peering and Sets
...
-writes to objects for a given placement group where it acts as the *Primary*.
-
-When a series of OSDs are responsible for a placement group, that series of
-OSDs, we refer to them as an *Acting Set*. An *Acting Set* may refer to the Ceph
-OSD Daemons that are currently responsible for the placement group, or the Ceph
-OSD Daemons that were responsible  for a particular placement group as of some
+(that is, ``size = 2``), which is the minimum requirement for data safety. For
+high availability, a Ceph Storage Cluster should store more than two copies of
+an object (that is, ``size = 3`` and ``min size = 2``) so that it can continue
+to run in a ``degraded`` state while maintaining data safety.
```
Comment (anthonyeleven) [inline doc/architecture.rst:633]: With R2 it's not a question *if* one will lose data, but *when*.  Maybe add a note that 2R for any data you care about is begging to lose.

File: doc/architecture.rst [hunk 5]
```diff
@@ -615,46 +615,46 @@ Peering and Sets
...
+
+As explained in the diagram in `Smart Daemons Enable Hyperscale`_, we do not
+name the Ceph OSD Daemons specifically (for example, ``osd.0``, ``osd.1``,
+etc.), but rather refer to them as *Primary*, *Secondary*, and so forth. By
+convention, the *Primary* is the first OSD in the *Acting Set*, and is
```
Comment (anthonyeleven) [inline doc/architecture.rst:638]: Is it the acting set, or the up set?  I can never keep those straight.

File: doc/architecture.rst [hunk 6]
```diff
@@ -615,46 +615,46 @@ Peering and Sets
...
+responsible for coordinating the peering process for each placement group where
```
Comment (anthonyeleven) [inline doc/architecture.rst:639]: s/coordinating/orchestrating/ because it sounds cooler.

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:619]: Accepted.

File: doc/architecture.rst (hunk 2 above)
Comment (zdover23) [reply doc/architecture.rst:620]: Accepted.

File: doc/architecture.rst (hunk 5 above)
Comment (zdover23) [reply doc/architecture.rst:638]: I'm not sure. I'll ask Josh tomorrow. We'll return to this in a month or so, to make sure it's not nonsense.

File: doc/architecture.rst (hunk 6 above)
Comment (zdover23) [reply doc/architecture.rst]: Accepted.

File: doc/architecture.rst (hunk 5 above)
Comment (jdurgin) [reply doc/architecture.rst:638]: we actually refer to them in the osdmap as acting primary (from the acting set) and up primary (from the up set). In this context, we'd be talking about the acting set, the current OSDs that are actually responsible for a PG.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53871 - Reef backport
https://github.com/ceph/ceph/pull/53872 - Quincy backport

**Pull request merged**